### DISCUSSION TITLE
Is SEI Layer growth temperature independent??

### DISCUSSION THREAD

**Discussion by mlnsai** [pybamm-team/PyBaMM#2992] [Q&A] [answered]
Hi Team,

I have been conducting simple cyclic testing simulations for 350 cycles at different temperatures. (15C, 30C, 45C)

Here is the cycle

```
    experiment = pybamm.Experiment(
        
        [
            "Charge at 1 C until 4.2 V",
            "Hold at 4.2 V until C/10",
            "Rest for 5 minutes",
            "Discharge at 1 C until 2.8 V",
            "Rest for 5 minutes",
        ]* 350
    )

```

Here are the options

```
model = pybamm.lithium_ion.DFN(options=
{"thermal": "x-lumped", 
"SEI":"solvent-diffusion limited", 
"SEI porosity change" : "true", 
"lithium plating": "reversible"
})
```

Interestingly, the results I have got indicates that SEI Layer growth is temperature independent. But literature suggests the temperature dependency on SEI Layer growth. Whys is this happening in PyBaMM?

Any suggestions @brosaplanella @tinosulzer ? :)

Here I have attached the result I have got. Also attached the spreadsheet for the data.
[SEI_Analysis_Final.xlsx](https://github.com/pybamm-team/PyBaMM/files/11581852/SEI_Analysis_Final.xlsx)
![Result](https://github.com/pybamm-team/PyBaMM/assets/103929419/84d43d0b-2172-4f3a-83c0-fc1ee63a8c7a)

**Comment by brosaplanella** [comment]
That's probably because the parameters you use for SEI do not depend on temperature, so the behaviour is not captured even if the temperature changes in the model.

**Comment by mlnsai** [comment]
Thanks @brosaplanella for the reply. I am using the default SEI parameters and I am not updating/calling any parameters specifically.

The model I selected is "solvent-diffusion limited" and it is the same result for "ec reaction limited" as far as temperature independency is concerned. I have further enabled "SEI porosity change" : "true" option.

It would be great if you could point out those parameters which can capture the SEI dependency on temperature. I would like to re-run incorporating this temperature impact on SEI layer growth.

**Reply by brosaplanella** [reply]
If you don't explicitly pass parameter values the model uses the default parameter values (you can get them doing `model.default_parameter_values`). I am not aware of any parameter sets fully capturing the temperature dependence of SEI growth, so you should double check the published literature.

**Reply by mlnsai** [reply]
Hi @brosaplanella , thanks for the suggestion.

I did pass "Chen" and "Ramadass" and "Ai2020"parameter sets as of now. These didn't capture the temperature dependency. If you have any suggestion about any parameters that can capture the temperature dependency, kindly let me know. 

We are also trying from our end. I will keep you posted.

**Reply by brosaplanella** [reply]
At least Chen2020, which is the one I am familiar, uses some literature default SEI parameters which probably don't capture temperature dependence well. I might look into parameterisation of such models in the near future, will keep you posted.

**Comment by mpegis** [comment]
If its solvent-diffusion limited, I imagine a T-dependent diffusivity of EC would solve your problem. Is your diffusivity T-dependent?

**Reply by mlnsai** [reply]
Hi, @mpegis, thanks for the reply. 

But "EC diffusivity [m2.s-1]": 2e-18," this is constant for all parameter sets in PyBaMM. Is there any equation that actually makes this EC diffusivity temperature dpendent?

Thanks in advance

**Reply by mpegis** [reply]
I believe you should be able to put in a T-dependent function for that parameter although I could be wrong. You could find a literature T-dependent NMR study (e.g. https://pubs.acs.org/doi/10.1021/je3003089) and start from there, although you could also just manually change it for each temperature.

I would say generally, the fact that pybamm can model T-dependent kinetics requires parameters which contain T-dependence such as exchange current density, Li ion diffusivity, conductivity/etc. So you would want to modify your SEI submodel to contain best guesses for what could be happening.

**Comment by mlnsai** [comment, marked as answer]
Hi @mpegis , It did worked by making the diffusivity a temperature dependent function which changes with ambient temperature. 

An Arrhenius dependent function that takes ambient temperature as input worked just fine. 

The temperature-dependent diffusivity property is descripted by this Arrhenius equation

![image](https://github.com/pybamm-team/PyBaMM/assets/103929419/6ef1f722-6ac5-4f5f-8719-b6c9ef901dcf)


Here's the change I have done in code.

If it's an EC reaction limited SEI submodel. 

```
    def ec_diffusivity(T_amb):
        D_sj = 2e-18,
        EaD = 4000
        arrhenius = exp(EaD / constants.R * (1 / 298.15 - 1 / T_amb))
        return D_sj * arrhenius
        parameter_values.update({"EC diffusivity [m2.s-1]":ec_diffusivity(tempK)})
```


If it's an Solvent-diffusion limited SEI submodel. 


```
    def sd_diffusivity(T_amb):
        D_sd = 2.5e-22
        EaD = 4000
        arrhenius = exp(EaD / constants.R * (1 / 298.15 - 1 / T_amb))
        return D_sd * arrhenius
    parameter_values.update({"Outer SEI solvent diffusivity [m2.s-1]":sd_diffusivity(tempK)})
```


Here's a snapshot of the output

![image](https://github.com/pybamm-team/PyBaMM/assets/103929419/54894e7f-65b1-41fd-a045-e6f49d21c898)


Thanks @mpegis and @brosaplanella  for timely inputs.

**Reply by Ansh-Lodhi** [reply]
I was going through the source code of SEI growth sub Model and I found this :
![image](https://github.com/pybamm-team/PyBaMM/assets/135127806/7c034619-8aae-40d1-92cf-8d75f1c62c45)

Apparently, SEI growth is already dependent on temperature. Can somebody confirm this?

**Reply by brosaplanella** [reply]
Yes, but it only captures the Arrhenius dependence. Plus, you need to pass a non-zero value for the activation energy, otherwise the effect is ignored.

**Comment by ganmad** [comment]
Hello,

How to introduce a temperature dependence on diffusivity locally on notebook? Recent updates are throwing an error. Please check below minimum example. Did I miss something? Thank you.


```
import pybamm
model = pybamm.lithium_ion.SPM({"cell geometry": "arbitrary","thermal": "lumped","SEI": "ec reaction limited","SEI film resistance":"distributed",
        "SEI porosity change": "true",})
param = pybamm.ParameterValues("OKane2022")


def ec_diffusivity(T_dim):
       
    D_sj = 2.5*10**(-22)
    EaD = 30000
    arrhenius = pybamm.exp(EaD / pybamm.constants.R * (1 / 298.15 - 1 / T_dim))
    return D_sj * arrhenius

param.update({"EC diffusivity [m2.s-1]":ec_diffusivity})

exp = pybamm.Experiment(["Discharge at 0.1C until 2.5 V"])
sim = pybamm.Simulation(model,parameter_values=param,experiment=exp,)
sim.solve()
```


TypeError: Cannot process parameter '<function ec_diffusivity at 0x7fc0f89e6170>'

**Reply by mlnsai** [reply]
Hi, try giving ambient temperature in kelvin (tempK) as the input to the function while calling

`param.update({"EC diffusivity [m2.s-1]":ec_diffusivity (tempK)})`

**Reply by ganmad** [reply]
Hi @mlnsai, Thanks for the response. 

Do you mean calling a variable like below?

```
tempK = model.variables(["Cell temperature [K]"])  or tempK = model.variables(["Ambient temperature [K]"])

param.update({"EC diffusivity [m2.s-1]":ec_diffusivity(tempK)})
```

**Reply by MirAbbasAli2A** [reply]
@ganmad, could you please tell me where you got the EaD of 30,000 from?

**Comment by ganmad** [comment]
Introducing temperature dependence is not always straight forward. I think one should introduce new functions as hinted in this answer: 
https://github.com/pybamm-team/PyBaMM/discussions/3103#discussioncomment-6377736

Need to check whether the parameter is defined using `Parameter` class or `FunctionParameter` class. I understand `EC diffusivity [m2.s-1] `is defined using Parameter class, hence we can't introduce temperature dependence locally.  I still have to figure out how to implement though. Let me know if I am wrong. Thanks.

**Reply by brosaplanella** [reply]
Yes, EC diffusivity currently is a constant, so not temperature dependent. Again, it should be fairly easy to fix this so feel free to try and tackle it. Let me know if you need any help.
